### DISCUSSION TITLE
Question about the `@RestrictTo` reasonings.

### DISCUSSION THREAD

**Discussion by weeklytyped** [ReactiveCircus/FlowBinding#130] [Q&A] [answered]
Hi, thanks for the great work!

I was trying to copy-paste the `TextView.textChanges()` and modify it to emit not only the text but also the meta data (start,before,couont).  But get warned by lint that there are several methods marked as `@RestrictTo(LIBRARY_GROUP)`. 
Would like to understand the reasoning of those annotation and the risk of using such method outside library? 

Best regards.

**Comment by ychescale9** [comment]
Hi. There's a [request](https://github.com/ReactiveCircus/FlowBinding/issues/108) for removing these restrictions.

This risk of using these APIs is the library in theory could change these APIs in a source-incompatible way.

Although since these APIs aren't using any `Experimental` APIs themselves anymore, I don't see why we can't remove these annotations to allow users implement their own bindings.

**Comment by ychescale9** [comment, marked as answer]
https://github.com/ReactiveCircus/FlowBinding/pull/131
